On Mon, Jan 25, 2016 at 03:51:01PM -0800, Tuomas Tynkkynen wrote:

> Probably the subversion problem was caused by GCC 5 (which closure-size had but no longer seems to have) which had some changes in that area.
> 

Some comment taken from my own unstaged commit:
```
Mysterious error "configure: error: failed to recognize APR_INT64_T_FMT on this platform" appeared,
usually it appears on OSX, not on linux. Openembedded's forum suggest
CPPFLAGS="-P" to fix it.

Short snippet from svn's autocrap illustrating problematic place:
140           if test "x$svn_cv_pycfmt_apr_int64_t" = "x"; then
141             AC_EGREP_CPP([MaTcHtHiS +\"lld\" +EnDeNd],
142                          [#include <apr.h>
143                           MaTcHtHiS APR_INT64_T_FMT EnDeNd],
144                          [svn_cv_pycfmt_apr_int64_t="L"])
145           fi

I think line numbering can smash this magic, so disable it with `-P`

```

--- a/pkgs/applications/window-managers/fvwm/default.nix
+++ b/pkgs/applications/window-managers/fvwm/default.nix
@@ -1,7 +1,7 @@
 { gestures ? false
 , stdenv, fetchurl, pkgconfig
 , cairo, fontconfig, freetype, libXft, libXcursor, libXinerama
-, libXpm, librsvg, libpng, fribidi, perl
+, libXpm, libXt, librsvg, libpng, fribidi, perl
 , libstroke ? null
 }:
 
@@ -17,7 +17,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     pkgconfig cairo fontconfig freetype
-    libXft libXcursor libXinerama libXpm
+    libXft libXcursor libXinerama libXpm libXt
     librsvg libpng fribidi perl
   ] ++ stdenv.lib.optional gestures libstroke;
 

--- a/pkgs/applications/window-managers/qtile/default.nix
+++ b/pkgs/applications/window-managers/qtile/default.nix
@@ -24,9 +24,9 @@ buildPythonPackage rec {
 
   postPatch = ''
     substituteInPlace libqtile/manager.py --subst-var-by out $out
-    substituteInPlace libqtile/pangocffi.py --subst-var-by glib ${pkgs.glib}
-    substituteInPlace libqtile/pangocffi.py --subst-var-by pango ${pkgs.pango}
-    substituteInPlace libqtile/xcursors.py --subst-var-by xcb-cursor ${pkgs.xorg.xcbutilcursor}
+    substituteInPlace libqtile/pangocffi.py --subst-var-by glib ${pkgs.glib.out}
+    substituteInPlace libqtile/pangocffi.py --subst-var-by pango ${pkgs.pango.out}
+    substituteInPlace libqtile/xcursors.py --subst-var-by xcb-cursor ${pkgs.xorg.xcbutilcursor.out}
   '';
 
   buildInputs = [ pkgs.pkgconfig pkgs.glib pkgs.xorg.libxcb pkgs.cairo pkgs.pango python27Packages.xcffib ];

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -2565,7 +2565,7 @@ in modules // {
 
     patchPhase = ''
       # Hardcode cairo library path
-      sed -e 's,ffi\.dlopen(,&"${pkgs.cairo}/lib/" + ,' -i cairocffi/__init__.py
+      sed -e 's,ffi\.dlopen(,&"${pkgs.cairo.out}/lib/" + ,' -i cairocffi/__init__.py
     '';
 
     meta = {
@@ -23731,7 +23731,7 @@ in modules // {
 
     patchPhase = ''
       # Hardcode cairo library path
-      sed -e 's,ffi\.dlopen(,&"${pkgs.xorg.libxcb}/lib/" + ,' -i xcffib/__init__.py
+      sed -e 's,ffi\.dlopen(,&"${pkgs.xorg.libxcb.out}/lib/" + ,' -i xcffib/__init__.py
     '';
 
     propagatedBuildInputs = [ self.cffi self.six ];

